src/main/java/org/jenkinsci/deprecatedusage/DeprecatedUnusedApiReport.java

```diff
@@ -0,0 +1,141 @@
+package org.jenkinsci.deprecatedusage;
+
+import org.json.JSONObject;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.Writer;
+import java.util.List;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+public class DeprecatedUnusedApiReport extends Report {
+    public DeprecatedUnusedApiReport(DeprecatedApi api, List<DeprecatedUsage> usages, File outputDir, String reportName) {
+        super(api, usages, outputDir, reportName);
+    }
+
+    protected void generateHtmlReport(Writer writer) throws IOException {
+        writer.append("<h1>Unused Deprecated APIs</h1>");
+
+        {
+            writer.append("<div class='classes'><h2>Classes</h2><ul>\n");
+            CLASSES:
+            for (String deprecatedClass : new TreeSet<>(api.getClasses())) {
+                if (!isRelevantSignature(deprecatedClass)) {
+                    continue;
+                }
+                for (DeprecatedUsage usage : usages) {
+                    if (usage.getClasses().contains(deprecatedClass)) {
+                        continue CLASSES;
```

Comment:
It's just a `continue` for the outer loop, the idea being that the inner loop determins whether the current iteration of the outer loop is relevant. This is easier than e.g. setting a flag and breaking.